src/ape/api/providers.py

```diff
@@ -1362,6 +1432,127 @@ def _find_txn_by_account_and_nonce(
                     stop_block,
                 )
 
+    def poll_blocks(
+        self,
+        stop_block: Optional[int] = None,
+        required_confirmations: Optional[int] = None,
+        new_block_timeout: Optional[int] = None,
+    ) -> Iterator[BlockAPI]:
+        # Wait half the time as the block time
+        # to get data faster.
+        block_time = self.network.block_time
+        wait_time = block_time / 2
+
+        # The timeout occurs when there is no chain activity
+        # after a certain time.
+        timeout = (
+            (10.0 if self.network.is_dev else 50 * block_time)
+            if new_block_timeout is None
+            else new_block_timeout
+        )
+
+        # Only yield confirmed blocks.
+        if required_confirmations is None:
+            required_confirmations = self.network.required_confirmations
+
+        @dataclass
+        class YieldAction:
+            hash: bytes
+            number: int
+            time: float
+
+        # Pretend we _did_ yield the last confirmed item, for logic's sake.
+        fake_last_block = self.get_block(self.web3.eth.block_number - required_confirmations)
+        last = YieldAction(
+            number=fake_last_block.number, hash=fake_last_block.hash, time=time.time()
+        )
+
+        # A helper method for various points of ensuring we didn't timeout.
+        def assert_chain_activity():
+            time_waiting = time.time() - last.time
+            if time_waiting > timeout:
+                raise ProviderError("Timed out waiting for next block.")
+
+        # Begin the daemon.
+        while True:
+            # The next block we want is simply 1 after the last.
+            next_block = last.number + 1
+
+            # Use an "adjused" head, based on the required confirmations.
+            head = self.get_block("latest")
+
+            try:
+                adjusted_head = self.get_block(head.number - required_confirmations)
+            except Exception:
+                # TODO: I did encounter this sometimes in a re-org, needs better handling
```

Comment:
@z80dev I hit this sometimes in reorg in some race condition, like a block-not-found error of some kind, where it reorgs in between getting the head and adjusted head